doc/Clock commands/clock_managment_hld.md

```diff
@@ -0,0 +1,215 @@
+		* Date & timezone setting will be reflected of all the services/dockers in the system.
+
+
+	2. config clock date <YYYY-MM-DD> <HH:MM:SS> (to set time and date)
+		* Command will recieve single string with date-time format <YYYY-MM-DD> <HH:MM:SS>
+		* Command does not need to be stored in DB.
+		* Linux timedatectl with set-time flag will be called.
+		e.g. timedatectl set-time "2012-10-30 18:17:16"
+		* <b>In case NTP is enabled -> time/date set is NOT allowed and being blocked</b>
+
+	Additional show command will be added to display all valid timezones:
+
+	3. show clock timezones
+
+		* Will display list of all valid timezones to be configured.
+		* based on "timedatectl list-timezones" linux command.
+
+
+<u>General notes:</u><br>
+
+	* Both set commands will be written directly to Linux (via imedatectl command), and will be activated immediately.
+
+	* In case of any set command (time/date/timezone) a relevant message will be print to syslog.
+
+	* In case of any set command (time/date/timezone) - rsyslog service should be restarted to reflect the time change.
+
+	* Upgrades - timezone configuration is part of DEVICE_METADATA, hence - configuration upgrade will be taken care of as any other sonic configuration.
+
+
+##  3 <a name='Configurationandmanagement'></a>Configuration and management
+
+###  3.1 <a name='ConfigDBTables'></a>ConfigDB Tables
+
+Only timezone configuration will be saved.
+Additional field will be added to DEVICE_METADATA table
+
+```
+DEVICE_METADATA :{
+    "timezone": {{string}}
+}
+```
+
+default value: "Etc/UTC"
+
+
+###  3.2 <a name='CLIYANGmodel'></a>CLI/YANG model
+
+####  3.2.1. <a name='Yangmodel'></a>Yang model
+
+
+Adding to existing Yang model of device_metadata (https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-device_metadata.yang)
+
+```
+    container sonic-device_metadata {
+
+        container DEVICE_METADATA {
+
+            description "DEVICE_METADATA part of config_db.json";
+
+            container localhost{
+			...
+				leaf timezone {
+					type string;
+					description "System time-zone setting";
+				}
+```
+
+####  3.2.2. <a name='Climodel'></a>CLI
+
+##### Show CLI
+
+```
+root@host:~$ show clock 
+Sun 15 Jan 2023 06:12:08 PM IST
+
+```
+
+```
+root@host:~$ show clock timezones
+Africa/Abidjan
+Africa/Accra
+Africa/Addis_Ababa
+Africa/Algiers
+Africa/Asmara
+...
+
+```
+
+##### Config CLI
+
+```
+root@host:~$ config clock timezone "<timezone>"
+
+```
+
+```
+root@host:~$ config clock date <YYYY-MM-DD> <HH:MM:SS>
+
+```
+
+
+##  4 <a name='TestPlan'></a>Test Plan
+
+###  4.1 <a name='UnitTestcases'></a>Unit Test cases
+
+1. Good flows:<br>
+	a. set timezone<br>
+	b. set date<br>
+	c. set time<br>
+	d. set date & time<br>
+	e. check reboot / upgrade<br>
+	f. check docker restart<br>
+	g. config load / reload <> <br>
+	h. reboot without config save<br>
+	i. upgrade with manually timedatectl before upgrade<br>
```

Comment:
Please add passing criteria, e.g:

After a change, during reboot, syslog generates continuous and consistent time/date in log entries. No time zone/clock change in between.

Reply:
Thanks for the comment, added.